### PR TITLE
Policylists

### PR DESCRIPTION
**Pull request by lawlawlaw**
Kept policylist refactoring, but removed references to mongodb.

### CODE REVIEW

File: src/Makefile.bench.include [hunk 1]
```diff
@@ -30,7 +30,7 @@ bench_bench_bitcoin_SOURCES = \
 nodist_bench_bench_bitcoin_SOURCES = $(GENERATED_TEST_FILES)
 
 bench_bench_bitcoin_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES) $(EVENT_CLFAGS) $(EVENT_PTHREADS_CFLAGS) -I$(builddir)/bench/
-bench_bench_bitcoin_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+bench_bench_bitcoin_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS) $(MONGOCXX_FLAGS)
```
Comment (nkostoulas) [inline src/Makefile.bench.include]: need to revert this

File: src/Makefile.bench.include [hunk 2]
```diff
@@ -52,7 +52,7 @@ bench_bench_bitcoin_LDADD += $(LIBBITCOIN_WALLET) $(LIBBITCOIN_CRYPTO)
 endif
 
 bench_bench_bitcoin_LDADD += $(BOOST_LIBS) $(BDB_LIBS) $(SSL_LIBS) $(CRYPTO_LIBS) $(MINIUPNPC_LIBS) $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS)
-bench_bench_bitcoin_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)
+bench_bench_bitcoin_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(MONGOCXX_LIBS)
```
Comment (nkostoulas) [inline src/Makefile.bench.include]: and this

File: src/Makefile.qt.include [hunk 3]
```diff
@@ -407,7 +407,7 @@ endif
 qt_elements_qt_LDADD += $(LIBBITCOIN_CLI) $(LIBBITCOIN_COMMON) $(LIBBITCOIN_UTIL) $(LIBBITCOIN_CONSENSUS) $(LIBBITCOIN_CRYPTO) $(LIBUNIVALUE) $(LIBLEVELDB) $(LIBMEMENV) \
   $(BOOST_LIBS) $(QT_LIBS) $(QT_DBUS_LIBS) $(QR_LIBS) $(PROTOBUF_LIBS) $(BDB_LIBS) $(SSL_LIBS) $(CRYPTO_LIBS) $(MINIUPNPC_LIBS) $(LIBSECP256K1) \
   $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS)
-qt_elements_qt_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(QT_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)
+qt_elements_qt_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(QT_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(MONGOCXX_LIBS)
```
Comment (nkostoulas) [inline src/Makefile.qt.include]: and these

File: src/Makefile.qttest.include [hunk 4]
```diff
@@ -48,8 +48,8 @@ qt_test_test_elements_qt_LDADD += $(LIBBITCOIN_CLI) $(LIBBITCOIN_COMMON) $(LIBBI
   $(LIBMEMENV) $(BOOST_LIBS) $(QT_DBUS_LIBS) $(QT_TEST_LIBS) $(QT_LIBS) \
   $(QR_LIBS) $(PROTOBUF_LIBS) $(BDB_LIBS) $(SSL_LIBS) $(CRYPTO_LIBS) $(MINIUPNPC_LIBS) $(LIBSECP256K1) \
   $(EVENT_PTHREADS_LIBS) $(EVENT_LIBS)
-qt_test_test_elements_qt_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(QT_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)
-qt_test_test_elements_qt_CXXFLAGS = $(AM_CXXFLAGS) $(QT_PIE_FLAGS)
+qt_test_test_elements_qt_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(QT_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(MONGOCXX_LIBS)
+qt_test_test_elements_qt_CXXFLAGS = $(AM_CXXFLAGS) $(QT_PIE_FLAGS) $(MONGOCXX_CFLAGS)
```
Comment (nkostoulas) [inline src/Makefile.qttest.include]: ++

File: src/Makefile.test.include [hunk 5]
```diff
@@ -182,7 +182,7 @@ test_test_elements_fuzzy_LDADD = \
   $(LIBBITCOIN_CRYPTO) \
   $(LIBSECP256K1)
 
-test_test_elements_fuzzy_LDADD += $(BOOST_LIBS) $(CRYPTO_LIBS)
+test_test_elements_fuzzy_LDADD += $(BOOST_LIBS) $(CRYPTO_LIBS) $(MONGOCXX_LIBS)
```
Comment (nkostoulas) [inline src/Makefile.test.include]: ++

File: src/rpc/protocol.h [hunk 6]
```diff
@@ -52,7 +52,8 @@ enum RPCErrorCode
     RPC_VERIFY_ALREADY_IN_CHAIN     = -27, //!< Transaction already in chain
     RPC_IN_WARMUP                   = -28, //!< Client still warming up
     RPC_INVALID_KEY_DERIVATION      = -32, //!< Invalid key derivation when tweaking key with contract hash
-
+    RPC_MONGOCXX_EXCEPTION          = -33, //!< mongocxx database exception
```
Comment (nkostoulas) [inline src/rpc/protocol.h]: ++

File: src/policy/whiteList.cpp [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+			    std::error_code(CPolicyList::Errc::INVALID_ADDRESS_OR_KEY, std::system_category()),
+			    std::string(__func__) + ": invalid key id");
+  //      throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid key id");
+  
+  if (pubKey.GetID() != keyId)
+    throw std::system_error(
+			    std::error_code(CPolicyList::Errc::INVALID_ADDRESS_OR_KEY,std::system_category()), 
+			    std::string(__func__) + std::string(": invalid key derivation when tweaking key with contract hash for tweaked address: ") 
+			    + addressIn + std::string(", public key ") + key );
+  //      throw JSONRPCError(RPC_INVALID_KEY_DERIVATION, string("Invalid key derivation when tweaking key with contract hash for tweaked address: ")
```
Comment (nkostoulas) [inline src/policy/whiteList.cpp]: ++

File: src/init.cpp [hunk 8]
```diff
@@ -903,6 +904,13 @@ bool AppInitBasicSetup()
     return true;
 }
 
+std::string GetEnvOrDefault(const std::string& var,
```
Comment (nkostoulas) [inline src/init.cpp]: Usually we set parameters equal to some env variable in Docker so no need to do this in code. 

But I think you used this for mongo stuff, which we removed, so can remove this.

File: src/policy/whiteList.cpp (hunk 7 above)
Comment (lawlawlaw) [reply src/policy/whiteList.cpp]: We need to keep this as it is.

Comment (nkostoulas) [reply src/policy/whiteList.cpp]: sorry just meant the commented stuff

**Pull request merged**